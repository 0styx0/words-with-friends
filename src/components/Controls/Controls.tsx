import * as React from 'react';
import Controls from './';

export default class ControlsContainer extends React.Component<{}, {}> {

    render(): any {
        return <Controls />;
    }
}