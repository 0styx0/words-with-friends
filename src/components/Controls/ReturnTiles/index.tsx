import * as React from 'react';

interface Props {
    returnTiles: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

export default function ReturnTiles(props: Props) {
    return <button onClick={props.returnTiles}>Return Tiles</button>;
}
