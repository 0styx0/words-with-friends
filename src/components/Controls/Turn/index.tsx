import * as React from 'react';

interface Props {
   turn: Function;
   number: number;
}

export default function Turn(props: Props) {

    return (
        <>
            <span className="white">Current Turn: {props.number}</span>
            <button id="playButton" onClick={() => props.turn()} type="button">Play</button>
        </>
   );
}
