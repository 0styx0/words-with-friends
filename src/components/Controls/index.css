#controls {
    background-color: green;
    max-width: 24vh;
    min-height: 10vh;
    padding: 5px;
    display: block;
    margin: 0 auto;
}

#controls button {
    display: block;
}

.white {
    color: tan;
}
