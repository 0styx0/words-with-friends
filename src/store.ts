import reducers from './reducers';
import { createStore, compose } from 'redux';
import Player from './classes/Player';
import Tilebag from './classes/Tilebag';
import Board from './classes/Board';

const enhancers = compose('devToolsExtension' in window ?
    (window as {} as { devToolsExtension: Function }).devToolsExtension!() :
    (f: {}) => f);

export const defaultState = {
    turn: 0,
    Players: [new Player(true, 0), new Player(false, 1)], // turns switched around on INIT_PLAYERS
    board: new Board(),
    Tilebag: new Tilebag()
};

const store = createStore(reducers, defaultState, enhancers);

export default store;
