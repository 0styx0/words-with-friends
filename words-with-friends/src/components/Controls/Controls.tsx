import * as React from 'react';
import Controls from './';

interface Props {
    turn: Function;
}

export default class ControlsContainer extends React.Component<Props, {}> {

    render() {
        return <Controls play={this.props.turn} />;
    }

}