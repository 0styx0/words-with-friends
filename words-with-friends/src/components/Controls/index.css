#controls {
    background-color: green;
    width: 10vw;
    height: 10vh;
    margin: 20px auto;
    display: block;
}