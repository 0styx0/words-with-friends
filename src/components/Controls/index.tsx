import * as React from 'react';
import './index.css';
import Turn from './Turn/Turn';

export default function Controls() {

    return <section id="controls"><Turn {...[] as any} /></section>;
}


