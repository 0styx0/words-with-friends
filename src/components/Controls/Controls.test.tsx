import * as React from 'react';
import Controls from './Controls';
import * as renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../store';

describe('<Controls />', () => {

    it('renders correctly', () => {

        const tree = renderer.create(
            <Provider store={store}>
                <Controls />
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});