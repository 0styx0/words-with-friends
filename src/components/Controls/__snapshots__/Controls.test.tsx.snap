// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> renders correctly 1`] = `
<div>
  <button
    id="controls"
    onClick={[Function]}
    type="button"
  >
    Play
  </button>
</div>
`;
